pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -270,3 +283,21 @@ func maybeResizeDisk(diskfile string, maxsizebytes uint64) error {
 	}
 	return nil
 }
+
+//createTargetVhost creates target and vhost for device using information from VolumeStatus
+//and returns wwn to use for mounting
+func createTargetVhost(device string, status *types.VolumeStatus) (string, error) {
+	serial := tgt.GenerateNaaSerial()
+	wwn := fmt.Sprintf("naa.%s", serial)
+	err := tgt.TargetCreateIBlock(device, status.Key(), serial)
+	if err != nil {
+		return "", fmt.Errorf("TargetCreateFileIODev(%s, %s, %s): %v",
+			device, status.Key(), serial, err)
+	}
+	err = tgt.VHostCreateIBlock(status.Key(), wwn)
+	if err != nil {
+		return "", fmt.Errorf("VHostCreateIBlock(%s, %s): %v",
```

Comment:
Shouldn't you delete the IBlock here so that nothing was created when there was a error?

Reply:
Correct, added